AlphaWallet/Core/Coordinators/WalletBalance/WalletBalanceFetcher.swift

```diff
@@ -155,7 +155,9 @@ class WalletBalanceFetcher: NSObject, WalletBalanceFetcherType {
 
         for each in servers {
             //NOTE: when we change servers it might happen the case when native
-            tokensDataStore.addEthToken(forServer: each)
+            queue.async {
```

Comment:
I take this one as a temporary fix? :) We should definitely not be doing this in the long run.